examples/with-remix-thirdpartyemailpassword/app/routes/_index.tsx

```diff
@@ -1,110 +1,203 @@
+}
 
 export default function Home() {
-  const session = useSessionContext();
+  const loaderData = useLoaderData<{
+    session: ExtendedSession | null;
+    error: Error | null;
+  }>();
 
-  if (session.loading) {
+  if (!loaderData) {
     return <div>Loading...</div>;
   }
+  if (loaderData.error) {
+    return (
+      <div>
+        Something went wrong while trying to get the session. Error -
+        {loaderData.error.message}
+      </div>
+    );
+  }
 
-  if (session.doesSessionExist === false) {
-    return <div>Session does not exist</div>;
+  if (!loaderData.session?.accessToken) {
+    return redirect("/auth");
   }
 
-  const data = {
-    note: "Retrieve authenticated user-specific data from your application post-verification through the use of the verifySession middleware.",
-    userId: session.userId,
-    sessionHandle: session.accessTokenPayload.sessionHandle,
-    accessTokenPayload: session.accessTokenPayload,
-  };
-
-  const displaySessionInformationWindow = () => {
-    window.alert("Session Information: " + JSON.stringify(data));
-};
-
-  const links: {
-    name: string,
-    link: string,
-    icon: string,
-  }[] = [
-    {
-      name: "Blogs",
-      link: "https://supertokens.com/blog",
-      icon: BlogsIcon,
-    },
-    {
-      name: "Guides",
-      link: recipeDetails.docsLink,
-      icon: GuideIcon,
-    },
-    {
-      name: "Sign Out",
-      link: "",
-      icon: SignOutIcon,
-    },
-  ];
-
-  return (
-    <div className="homeContainer">
-      <div className="mainContainer">
-        <div className="topBand successTitle bold500">
-          <img
-            src={CelebrateIcon}
-            alt="Login successful"
-            className="successIcon"
-          />
-          Login successful
-        </div>
-        <div className="innerContent">
-          <div>Your userID is: </div>
+  if (loaderData.session) {
+    const sessionData: SessionDataForUI = {
+      note: "Retrieve authenticated user-specific data from your application post-verification through the use of the verifySession middleware.",
+      userId: loaderData.session.userId,
+      sessionHandle: loaderData.session.userDataInAccessToken.sessionHandle,
+      accessTokenPayload: loaderData.session.userDataInAccessToken,
+    };
+
+    const displaySessionInformationWindow = (sessionData: SessionDataForUI) => {
+      window.alert("Session Information: " + JSON.stringify(sessionData));
+    };
+
+    const links: {
+      name: string;
+      link: string;
+      icon: string;
+    }[] = [
+      {
+        name: "Blogs",
+        link: "https://supertokens.com/blog",
+        icon: BlogsIcon,
+      },
+      {
+        name: "Guides",
+        link: recipeDetails.docsLink,
+        icon: GuideIcon,
+      },
+      {
+        name: "Sign Out",
+        link: "",
+        icon: SignOutIcon,
+      },
+    ];
 
-          <div className="truncate userId">{session.userId}</div>
+    return (
+      <div className="homeContainer">
```

Comment:
this needs to be wrapped around in <SessionAuth> component from our react SDK. See the nextjs integration please!

Reply:
It is now wrapped in a `<SessionAuthForRemix>` component.